recipes/ginkgo/all/conanfile.py

```diff
@@ -101,8 +101,58 @@ def build_requirements(self):
     def source(self):
         get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
+    def _get_visual_generator(self):
+        compiler_version = str(self.settings.compiler.version)
+
+        if self.settings.compiler == "msvc":
+            toolset_override = self.conf.get("tools.microsoft.msbuild:vs_version", check_type=str)
+            if toolset_override:
+                visual_version = toolset_override
+            else:
+                visual_version = {
+                    "170": "11",
+                    "180": "12",
+                    "190": "14",
+                    "191": "15",
+                    "192": "16",
+                    "193": "17",
+                }[compiler_version]
+        else:
+            visual_version = compiler_version
+
+        visual_gen_suffix = {
+            "8": "8 2005",
+            "9": "9 2008",
+            "10": "10 2010",
+            "11": "11 2012",
+            "12": "12 2013",
+            "14": "14 2015",
+            "15": "15 2017",
+            "16": "16 2019",
+            "17": "17 2022",
+        }[visual_version]
+
+        return f"Visual Studio {visual_gen_suffix}"
+
     def generate(self):
         tc = CMakeToolchain(self)
+
+        # Since 1.5.0, force MSBuild generator if:
+        # - msvc & shared
+        # - user has set tools.cmake.cmaketoolchain:generator to Ninja
+        # It's a workaround for LINK : fatal error LNK1189: library limit of 65535 objects exceeded
+        # See https://github.com/ginkgo-project/ginkgo/issues/1495#issuecomment-1841983036
+        # and https://github.com/microsoft/vcpkg/pull/34280#discussion_r1351515284
+        if Version(self.version) >= "1.5.0" and is_msvc(self) and self.options.shared:
+            if self.conf.get("tools.cmake.cmaketoolchain:generator") == "Ninja":
+                visual_generator = self._get_visual_generator()
+                self.output.warning(
+                    f"You asked for Ninja generator, but fallback to {visual_generator}, "
+                    "otherwise it will fail during creation of DLL with "
+                    "LINK : fatal error LNK1189: library limit of 65535 objects exceeded"
+                )
```

Comment:
Thinking about it now, this might be confussing behaviour - were there any objetctions to raising in `validate()` for this case?

Reply:
As a consumer I consider that CMake generator is a wish for an internal detail and has no impact on produced binaries, so I don't care if it's not honored. I prefer a simple conan install command so that it can work out of the box, instead of "die and retry" iterations in order to find correct conan command to install my dependencies.

So sure we could raise, but I believe it would be a bad user experience.

Reply:
see also https://github.com/conan-io/conan/issues/12669#issuecomment-1868397919